web/admin/application/controllers/KlearCustomRunCodeController.php

```diff
@@ -64,11 +65,21 @@ public function runSpecificCodeAction()
                     );
                     $this->view->language = $language->toDto();
 
+                    /** @var \Ivoz\Provider\Domain\Model\Extension\Extension $extension */
+                    $extension = $this->dataGateway->remoteProcedureCall(
+                        \Ivoz\Provider\Domain\Model\User\User::class,
+                        $userModel->getId(),
+                        'getExtension',
+                        []
+                    );
+
                     /**
                      * For backward compatibility reasons
                      * @deprecated this will be remove in ivozprovider 3.0
                      */
-                    $this->view->user->setLanguage($this->view->language);
+                    if ($extension) {
```

Comment:
We should add a data verification logic to ensure the terminal has a User and that user has SceenExtension assigned, throwing an exception otherwise.

That should make this checks unnecessary.